types/mui-datatables/index.d.ts

```diff
@@ -103,12 +103,18 @@ export interface MUIDataTableTextLabels {
     selectedRows: MUIDataTableTextLabelsSelectedRows;
 }
 
+export interface MUIDataTableFilterOptions {
+    names?: string[];
+    display?: (filterList: string[], onChange: any, index: number, column: any) => void;
+    logic?: any;
+}
+
 export interface MUIDataTableColumnOptions {
     display?: Display;
     empty?: boolean;
     filter?: boolean;
     filterList?: string[];
-    filterOptions?: string[];
+    filterOptions?: MUIDataTableFilterOptions;
```

Comment:
Is it correct that this is no longer an array?

Reply:
Yes, see this [example](https://github.com/gregnb/mui-datatables/blob/master/examples/column-filters/index.js) for details, also mentioned on in the [readme](https://github.com/gregnb/mui-datatables). The string array with options moved into an object i called MUIDataTableFilterOptions with new settings `display ` and `logic`. As said: I'm not sure about the exact typing of especially display.

Reply:
I see em, suggested a better typing for `logic` below. I don't think much more can be done for `display` unless they properly document their typings.